airflow/contrib/hooks/gcp_dlp_hook.py

```diff
@@ -0,0 +1,1362 @@
+        """
+
+        client = self.get_conn()
+
+        if not name:
+            raise AirflowException(
+                "Please provide the triggeredJob name to be deleted."
+            )
+
+        client.delete_job_trigger(
+            name=name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def delete_stored_info_type(self, name, retry=None, timeout=None, metadata=None):
+        """
+        Deletes a stored infoType.
+
+        :param name: Resource name of the organization and storedInfoType to be deleted.
+        :type name: str
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: sequence[tuple[str, str]]]
+        """
+
+        client = self.get_conn()
+
+        if not name:
+            raise AirflowException(
+                "Please provide the storedInfoType name to be deleted."
+            )
+
+        client.delete_stored_info_type(
+            name=name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def get_deidentify_template(self, name, retry=None, timeout=None, metadata=None):
+        """
+        Gets a DeidentifyTemplate.
+
+        :param name: Resource name of the organization and deidentify template to be read.
+        :type name: str
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: sequence[tuple[str, str]]]
+        :rtype: google.cloud.dlp_v2.types.DeidentifyTemplate
+        """
+
+        client = self.get_conn()
+
+        if not name:
+            raise AirflowException("Please provide the deidentifyTemplate name.")
+
+        return client.get_deidentify_template(
+            name=name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def get_dlp_job(self, name, retry=None, timeout=None, metadata=None):
+        """
+        Gets the latest state of a long-running DlpJob.
+
+        :param name: The name of the DlpJob resource.
```

Comment:
It is worth adding the possibility of defining the project ID using the connection configuration.